components/brave_wallet/browser/cardano/cardano_transaction_unittest.cc

```diff
@@ -0,0 +1,232 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/cardano/cardano_transaction.h"
+
+#include <utility>
+
+#include "base/strings/string_number_conversions.h"
+#include "brave/components/brave_wallet/browser/cardano/cardano_rpc_schema.h"
+#include "brave/components/brave_wallet/common/test_utils.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+using testing::_;
+
+namespace brave_wallet {
+namespace {
+constexpr char kTxid1[] =
```

Comment:
same as the comment below, these are very opaque and the naming is not helpful either (txid1, txid2, etc..). Also I'm not seeing checks for edge cases like invalid transactions that you have checks for in CardanoTransaction. This applies to all code, but in particular we're dealing with people's money here and these things need to be tested exhaustively.